pkg/k8s/watchers/watcher.go

```diff
@@ -812,6 +816,10 @@ func genCartesianProduct(
 	for fePortName, fePort := range ports {
 		var besValues []*loadbalancer.Backend
 		for addrCluster, backend := range bes.Backends {
+			// Skip backends that are not having the same IP as frontend IP for headless service
```

Comment:
This is not clear to me because the comment tells me what the code is doing but it doesn't tell why we are doing this.

Reply:
Thanks, I have updated the comment with example, hopefully it will be clearer now.